tests/integration/test_vulnerability_detector/test_providers/test_providers_os.py

```diff
@@ -130,39 +99,111 @@ def test_providers(get_configuration, configure_environment, restart_modulesd):
     wazuh_min_version: 4.2.0
 
     parameters:
-        - get_configuration:
+        - configuration:
+            type: dict
+            brief: Wazuh configuration data. Needed for set_wazuh_configuration fixture.
+        - metadata:
+            type: dict
+            brief: Wazuh configuration metadata
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set the wazuh configuration according to the configuration data.
+        - truncate_log_files:
+            type: fixture
+            brief: Truncate the log files at the end of the testing case.
+        - clean_cve_tables_func:
+            type: fixture
+            brief: Clean all the CVE tables before and after running the test.
+        - restart_modulesd_function:
+            type: fixture
+            brief: Restart the wazuh-modulesd daemon.
+
+    assertions:
+        - The provider os data update starts.
+        - The warning message appears when `os` tag is empty in NVD and MSU providers.
+
+    input_description:
+        - The `test_os.yaml` file provides the module configuration for this test.
+
+    expected_output:
+        - r'Invalid option 'os' for '{provider_name}' provider.*'
+        - 'Starting '{provider_name}' database update'
+    '''
+    provider_name = metadata['provider_name']
+    operating_system = metadata['os']
+    if operating_system == '':
```

Comment:
```suggestion

    if operating_system == '':
```

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/2549/commits/bcfe82b71c1952681de4c8862a2335609cd102b4